akka-stream/src/main/scala/akka/stream/impl/io/TLSActor.scala

```diff
@@ -292,6 +292,7 @@ class TLSActor(
       if (tracing) log.debug("closing inbound")
       try engine.closeInbound()
       catch { case ex: SSLException ⇒ outputBunch.enqueue(UserOut, SessionTruncated) }
+      lastHandshakeStatus = engine.getHandshakeStatus
```

Comment:
I'm not sure if we ever need engine.closeInbound a few lines below.  After a SSLException the engine will be in the closed state already and `engine.closeInbound()` will generate a new `SSLException`. This exception will be handled by the `Pump#pump` through `pumpFailed()` which will invoke `fail` second time.